.github/workflows/external-contributor.yml

```diff
@@ -11,12 +12,17 @@ jobs:
     runs-on: ubuntu-latest
     if: github.event.pull_request.head.repo.full_name != github.repository
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install Python dependencies
```

Comment:
you can maybe use [setup-python](https://github.com/actions/setup-python#setup-python) here which can cache the dependencies (I think we must write it [like this](https://github.com/DataDog/datadog-agent/pull/24338/files#diff-76ad82686036d63b3bf52974dddfbe8fc72cd4172a0085d500190a3adcca3b41R30) to make it work)